app.py

```diff
@@ -1,261 +1,292 @@
 import streamlit as st
-import plotly.express as px
 from policyengine_us import Simulation
-from policyengine_core.charts import format_fig
 from policyengine_us.variables.household.demographic.geographic.state_code import (
     StateCode,
 )
-from policyengine_us.variables.household.income.household.household_benefits import household_benefits as HouseholdBenefits
-import numpy as np
+from policyengine_us.variables.household.income.household.household_benefits import (
+    household_benefits as HouseholdBenefits,
+)
+from policyengine_us.variables.household.income.household.household_tax_before_refundable_credits import (household_tax_before_refundable_credits as HouseholdTaxBeforeRefundableCredits,)
+
 import pandas as pd
-import hashlib
-# Create a function to get net income for the household, married or separate.
-def get_heatmap_values(state_code, children_ages, tax_unit):
-    # Tuple of net income for separate and married.
-    net_income_married = get_marital_values(
-        state_code, True, children_ages, tax_unit
-    )
-    net_income_separate = get_marital_values(state_code,False,children_ages, tax_unit)
-    final_separate = []
-    for val in net_income_separate:
-        temp_array = []
-        for val2 in net_income_separate:
-            temp_array.append(val + val2)
-        final_separate.append(temp_array)
 
-    return net_income_married, final_separate
+import yaml
+import pkg_resources
+import datetime
+
 
+def load_credits_from_yaml(package, resource_path):
+    yaml_file = pkg_resources.resource_stream(package, resource_path)
+    data = yaml.safe_load(yaml_file)
+    # Find the newest available year
+    newest_year = max(data['values'].keys())
+    credits = data['values'].get(newest_year, [])
+
+    return credits
+
+
+# Constants
 DEFAULT_AGE = 40
 YEAR = "2024"
-HEAT_MAP_OUTPUTS = {
-    "Income": ["household_net_income","employment_income"],
-    "Benefits":["household_benefits", "employment_income"],
-    "Taxes": ["household_tax_before_refundable_credits","employment_income"] , 
-    "Credits": ["household_refundable_tax_credits", "employment_income"]
+
+# Streamlit heading and description
+st.header("Marriage Incentive Calculator")
+st.write(
+    "This application evaluates marriage penalties and bonuses of couples, based on state and individual employment income"
+)
+st.markdown(
+    "This application utilizes <a href='https://github.com/PolicyEngine/us-marriage-incentive'>the policyengine API</a>",
+    unsafe_allow_html=True,
+)
+
+# Streamlit inputs for state code, head income, and spouse income.
+statecodes = [s.value for s in StateCode]
+us_territories = {
```

Comment:
capitalize constants
```suggestion
US_TERRITORIES = {
```